client/state/utils.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import tv4 from 'tv4';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+
+/**
+ * Module variables
+ */
+
+export function isValidStateWithSchema( state, schema, checkForCycles = false, banUnknownProperties = false ) {
+	const result = tv4.validateResult( state, schema, checkForCycles, banUnknownProperties );
+	if ( ! result.valid && config( 'env' ) === 'development' ) {
```

Comment:
I went ahead with the mockery setup for now. I might play around with sinon sandbox later so we can have something simpler.